pytorch_lightning/metrics/functional/sequence.py

```diff
@@ -0,0 +1,85 @@
+from collections import Counter
+
+import torch
+
+
+def _count_ngram(ngram_input_list: list, n_gram: int) -> Counter:
+    """Counting how many times each word appears in a given text with ngram
+
+    Args:
+        ngram_input_list: A list of translated text or reference texts
+        n_gram: gram value ranged 1 to 4
+
+    Return:
+        ngram_counter: a collections.Counter object of ngram
+    """
+
+    ngram_counter = Counter()
+
+    for i in range(1, n_gram + 1):
+        for j in range(len(ngram_input_list) - i + 1):
+            ngram_key = tuple(ngram_input_list[j : i + j])
+            ngram_counter[ngram_key] += 1
+
+    return ngram_counter
+
+
+def bleu_score(translate_corpus: list, reference_corpus: list, n_gram: int = 4, smooth: bool = False) -> torch.Tensor:
+    """Calculate BLEU score of machine translated text with one or more references.
+
+    Args:
+        translate_corpus: A list of lists of translated texts
+        reference_corpus: A list of lists of reference texts
+        n_gram: Gram value ranged from 1 to 4 (Default 4)
+        smooth: Whether or not to apply smoothing – Lin et al. 2004
+
+    Return:
+        A Tensor with BLEU Score
+
+    Example:
+
+        >>> translate_corpus = ["the FAST brown fox jumped over the lazy dog".split(' ')]
+        >>> reference_corpus = [["the quick brown fox jumped over the lazy dog".split(' '), "the quick brown fox jumped over the the lazy cat".split(' ')]]
+        >>> bleu_score(translate_corpus, reference_corpus)
+        tensor(0.7506)
+    """
+
+    assert len(translate_corpus) == len(reference_corpus)
+    numerator = torch.zeros(n_gram)
+    denominator = torch.zeros(n_gram)
+    precision_scores = torch.zeros(n_gram)
+    c = 0.0
+    r = 0.0
+    # referenced from https://pytorch.org/text/_modules/torchtext/data/metrics.html#bleu_score
```

Comment:
just saw this comment. If there is already a metric implemented in torch, why don't we just import this with an optional import here? Torchtext is part of the Pytorch environment, so this should be find, right @Borda ?

Reply:
The implementation is almost similar, the only differences are 
- current implementation includes `smooth` argument, torchtext doesn't
- current implementation returns `Tensor`, torchtext returns normal `float`

Reply:
Torchtext is extras so we can make option I am fine with that =)

Reply:
So, shall I refactor with torchtext?

Reply:
If we refactor it with torchtext, users without it cannot use this metric, but this metric is specific for text so they most likely have torchtext anyway... So let's refactor it...
Cc: @justusschock